server/testutils/factories/booking.ts

```diff
@@ -22,40 +22,57 @@ class BookingFactory extends Factory<Booking> {
       departureDate: future(),
       status: 'provisional',
       extensions: [],
+      confirmation: null,
+      arrival: null,
+      departure: null,
+      departures: [],
+      cancellation: null,
+      cancellations: [],
     })
   }
 
   confirmed() {
-    return this.params({
-      arrivalDate: soon(),
-      departureDate: future(),
+    return this.provisional().params({
       status: 'confirmed',
-      extensions: [],
+      confirmation: confirmationFactory.build(),
     })
   }
 
   arrived() {
```

Comment:
Where do you have in mind? I'm not sure where you're thinking of?

Reply:
ah just that when I do a search there are some tests across the codebase where we're still explicitly setting statuses - maybe we could refactor those to use these methods?

Reply:
The only cases I could see were things like

```
    const departedBooking = bookingFactory.build({
      ...this.booking,
      status: 'departed',
      departureDate: newDeparture.dateTime,
      departure,
    })
```

where we can't really use bookingFactory.departed(). Are there any others I've missed?

Reply:
You could do something like:
```
    const departedBooking = bookingFactory.departed().build({
      ...this.booking,
      departureDate: newDeparture.dateTime,
      departure,
    })
```

but then the `departed` status would get overwritten with the existing status from `this.booking`. I could change the signature of `departed` to take in an existing booking and use that, but then I don't think it's obvious which fields it'd take from the passed-in booking, and which it would overwrite

Reply:
ah right yeah, ok. That makes sense. Agree, that sounds like it would complicate things and not be worth the benefit.